controller/proxy-injector/webhook.go

```diff
@@ -166,6 +166,13 @@ func (w *Webhook) inject(request *admissionv1beta1.AdmissionRequest) (*admission
 		}
 		patch.addVolume(caBundle)
 		patch.addVolume(tlsSecrets)
+		patch.addPodAnnotations(map[string]string{
+			k8sPkg.IdentityModeAnnotation: k8sPkg.IdentityModeOptional,
```

Comment:
i'm not seeing `linkerd.io/identity-mode` show up on auto-injected pods. i'm wondering if `patch.addPodAnnotations` is blowing away previous calls, rather than adding them? note that we only call `addPodLabels` and similar once, and while we call `addVolume` twice, it's implemented with a `/spec/template/spec/volumes/-`, and first requires calling `addVolume`

repro:

```bash
$ bin/linkerd install --tls optional --proxy-auto-inject

$ curl  curl https://run.linkerd.io/emojivoto.yml > emojivoto.yml
# add linkerd.io/inject: enabled annotation to emojivoto namespace
$ cat emojivoto.yml | kubectl apply

$ kubectl -n emojivoto get po -o jsonpath='{.items[*].metadata.annotations}'
map[linkerd.io/created-by:linkerd/proxy-injector git-9a915b45 linkerd.io/proxy-version:git-9a915b45] map[linkerd.io/created-by:linkerd/proxy-injector git-9a915b45 linkerd.io/proxy-version:git-9a915b45] map[linkerd.io/created-by:linkerd/proxy-injector git-9a915b45 linkerd.io/proxy-version:git-9a915b45 voting-pod-foo:voting-pod-bar] map[linkerd.io/created-by:linkerd/proxy-injector git-9a915b45 linkerd.io/proxy-version:git-9a915b45]
```

Reply:
@siggy That's correct. I fixed that in #2334. I'll try to extract a diff from there that can be applied here.

Reply:
Not the same I did in the other branch, but this will do for this one:
https://gist.github.com/alpeb/2ad3f5df7038eb2822a7901c18db4b34